node/core/dispute-coordinator/src/initialized.rs

```diff
@@ -869,12 +869,21 @@ impl Initialized {
 			}
 		}
 
+		let has_own_vote = new_state.has_own_vote();
+		let is_disputed = new_state.is_disputed();
+		let has_controlled_indices = env.controlled_indices().is_empty();
+		let is_backed = self.scraper.is_candidate_backed(&candidate_hash);
+		let is_confirmed = new_state.is_confirmed();
+		// Refrain from participation if the dispute is not included, not backed and not confirmed
+		let refrain_participation = !is_included && !is_backed && !is_confirmed;
+
 		// Participate in dispute if we did not cast a vote before and actually have keys to cast a
-		// local vote:
-		if !new_state.has_own_vote() &&
-			new_state.is_disputed() &&
-			!env.controlled_indices().is_empty()
-		{
+		// local vote. Disputes should fall in one of the categories below, otherwise we will refrain
+		// from participation:
+		// - `is_included` lands in prioritised queue
+		// - `is_confirmed` | `is_backed` lands in best effort queue
+		// We don't participate in disputes on finalized candidates.
+		if !has_own_vote && is_disputed && !has_controlled_indices && !refrain_participation {
```

Comment:
We want to participate if we have controlled indices. Assuming the has_controlled_indices contents were fixed per the above comment, we want to remove the ! here.

Reply:
I think the code becomes a tad easier to read, if we remove the double negation. So instead of having a refrain_participation, make a negated `allow_participation`:

```rust
let allow_participation = is_included || is_backed || is_confirmed;
```